web/src/components/TaskHeader.test.tsx

```diff
@@ -192,6 +193,456 @@ describe("<TaskHeader />", () => {
+      selectCompleted();
+
+      expect(screen.queryByText("New Jersey Tax ID")).not.toBeInTheDocument();
+    });
+
+    it("shows ownership and employee count for non sp/gp", () => {
+      const taxTask = generateTask({ id: taxTaskId, stepNumber: 1 });
+
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          businessName: "",
+          legalStructureId: "limited-liability-company",
+        }),
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+        preferences: generatePreferences({ roadmapOpenSections: ["PLAN"] }),
+      });
+
+      useMockRoadmap({
+        steps: [generateStep({ stepNumber: 1, section: "PLAN" })],
+        tasks: [taxTask],
+      });
+
+      renderTaskHeader(taxTask, userData);
+      selectCompleted();
+
+      expect(screen.getByText(Config.profileDefaults.STARTING.ownership.header)).toBeInTheDocument();
+      expect(
+        screen.getByText(markdownToText(Config.profileDefaults.STARTING.existingEmployees.header))
+      ).toBeInTheDocument();
+    });
+
+    it("shows ownership and employee count for sp", () => {
+      const taxTask = generateTask({ id: taxTaskId, stepNumber: 1 });
+
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          businessName: "",
+          legalStructureId: "sole-proprietorship",
+        }),
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+        preferences: generatePreferences({ roadmapOpenSections: ["PLAN"] }),
+      });
+
+      useMockRoadmap({
+        steps: [generateStep({ stepNumber: 1, section: "PLAN" })],
+        tasks: [taxTask],
+      });
+
+      renderTaskHeader(taxTask, userData);
+      selectCompleted();
+
+      expect(screen.getByText(Config.profileDefaults.STARTING.ownership.header)).toBeInTheDocument();
+      expect(
+        screen.getByText(markdownToText(Config.profileDefaults.STARTING.existingEmployees.header))
+      ).toBeInTheDocument();
+    });
+
+    it("shows ownership and employee count for gp", () => {
+      const taxTask = generateTask({ id: taxTaskId, stepNumber: 1 });
+
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          businessName: "",
+          legalStructureId: "general-partnership",
+        }),
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+        preferences: generatePreferences({ roadmapOpenSections: ["PLAN"] }),
+      });
+
+      useMockRoadmap({
+        steps: [generateStep({ stepNumber: 1, section: "PLAN" })],
+        tasks: [taxTask],
+      });
+
+      renderTaskHeader(taxTask, userData);
+      selectCompleted();
+
+      expect(screen.getByText(Config.profileDefaults.STARTING.ownership.header)).toBeInTheDocument();
+      expect(
+        screen.getByText(markdownToText(Config.profileDefaults.STARTING.existingEmployees.header))
+      ).toBeInTheDocument();
+    });
```

Comment:
okay, so we have a *lot* of repetition in these last few tests. I think that for the goal of "test should be readable documentation of the thing's behavior" then we could male some refactoring here to make these more readable. I think making them more concise would be the path here.

here's the tests you have that are all very closely related:
- shows the business name field if it's not populated for non sp/gp
- doesn't show the business name field if it's populated for non sp/gp
- doesn't show the business name field for sp
- doesn't show the business name field for gp
- shows the tax id field for non sp/gp
- doesn't show the tax id field for sp
- doesn't show the tax id field for gp
- shows ownership and employee count for non sp/gp
- shows ownership and employee count for sp
- shows ownership and employee count for gp

For one, I think you could make use of random test generators to avoid duplicating tests for sp and for gp. Imagine if you had a function for `randomSpOrGpLegalStructure` and `randomNonSpGpLegalStructure`. Then you avoid duplication while still covering both the SP and GP cases.

If you had those two functions, it makes it clear that our tests break down into 2 behavior categories: SP/GP and non-SP/GP. We could make that explicit in the tests using describe blocks

Imagine something like:
```
describe('when SP/GP', () => {
 it "shows all fields except business name"
    expect(not business-name-field)
    expect(tax-id-field) 
    expect(ownership-field)
    expect(employee-count-field)

}

describe('when not SP/GP', () => {
  it "shows all fields except tax ID when business name not populated"
    expect(business-name-field)
    expect(not tax-id-field) 
    expect(ownership-field)
    expect(employee-count-field)

  it "does not show business name field when already populated"
    expect(not business-name-field)
}
```

obviously the names are not perfect but this is just one example breakdown that, at a high level, is readable so that people could actually reasonably look through the tests and be expected to understand what's actually going on

Reply:
Also, I matched your existing terminology here, but as I noted below, we should be using the domain language for these terms. It's not SP/GP and non-SP/GP, its Public Filing legal structure or Trade Name legal structure